Library/Formula/hbase.rb

```diff
@@ -4,32 +4,134 @@ class Hbase < Formula
   url "https://www.apache.org/dyn/closer.cgi?path=hbase/1.1.2/hbase-1.1.2-bin.tar.gz"
   sha256 "8ca5bf0203cef86b4a0acbba89afcd5977488ebc73eec097e93c592b16f8bede"
 
-  bottle :unneeded
-
   depends_on :java => "1.6+"
-  depends_on "hadoop"
+  depends_on "hadoop" => :optional
+  depends_on "lzo" => :recommended
+  depends_on "ant" => :build if build.with? "lzo"
+
+  resource "hadoop-lzo" do
+    url "https://github.com/cloudera/hadoop-lzo/archive/0.4.14.tar.gz"
+    sha256 "aa8ddbb8b3f9e1c4b8cc3523486acdb7841cd97c002a9f2959c5b320c7bb0e6c"
+  end
 
   def install
     rm_f Dir["bin/*.cmd", "conf/*.cmd"]
     libexec.install %w[bin conf docs lib hbase-webapps]
     bin.write_exec_script Dir["#{libexec}/bin/*"]
 
-    inreplace "#{libexec}/conf/hbase-env.sh",
-      "# export JAVA_HOME=/usr/java/jdk1.6.0/",
-      "export JAVA_HOME=\"$(/usr/libexec/java_home)\""
+    if build.with? "lzo"
+      resource("hadoop-lzo").stage do
+        # Fixed upstream: https://github.com/cloudera/hadoop-lzo/blob/master/build.xml#L235
+        inreplace "build.xml",
+                  %r{(<class name="com.hadoop.compression.lzo.LzoDecompressor" />)},
+                  "\\1\n<classpath refid=\"classpath\"/>"
+        ENV["CLASSPATH"] = "#{libexec}/lib/hadoop-common-*.jar"
+        ENV["CFLAGS"] = "-m64"
+        ENV["CXXFLAGS"] = "-m64"
+        ENV["CPPFLAGS"] = "-I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers"
+        system "ant", "compile-native", "tar"
+        (libexec/"lib").install Dir["build/hadoop-lzo-*/hadoop-lzo-*.jar"]
+        (libexec/"lib/native").install Dir["build/hadoop-lzo-*/lib/native/*"]
+      end
+    end
+
+    inreplace "#{libexec}/conf/hbase-env.sh" do |s|
+      # upstream bugs for ipv6 incompatibility:
+      # https://issues.apache.org/jira/browse/HADOOP-8568
+      # https://issues.apache.org/jira/browse/HADOOP-3619
+      s.gsub!("export HBASE_OPTS=\"-XX:+UseConcMarkSweepGC\"",
+              "export HBASE_OPTS=\"-Djava.net.preferIPv4Stack=true -XX:+UseConcMarkSweepGC\"")
+      s.gsub!("# export JAVA_HOME=/usr/java/jdk1.6.0/",
+              "export JAVA_HOME=\"$(/usr/libexec/java_home)\"")
+    end
+
+    inreplace "#{libexec}/conf/hbase-site.xml",
+              # makes hbase usable out of the box
+              # upstream has been provided this patch
+              # https://issues.apache.org/jira/browse/HBASE-15426
```

Comment:
Sorry for the pedantry but this indentation still feels weird. Can you make it two characters further than `inreplace` rather than aligning with the `"`? Thanks!